### ISSUE TITLE
PyTorch: string transformations

### ISSUE THREAD

**Issue by NennoMP** [gc-os-ai/pyaptamer#171]
As highlighted in #106, we need to structure the string transformations currently present in the package under a consistent interface compatible with `torch` dataloaders.

We should implement these transformations under `torch` core. There is not an established interface as transformations are generally contained elsewhere, e.g. `torchtext` or `torchvision` for text and images, respectively. Thus, rather than inheritance, the design is currently organized with duck typing - transforms should implement the `__call__` method in order to work with `nn.Sequential`. Note sure whether it is better to define an abstract class for the interface or stick with duck typing though.

Unfortunatelty `torchtext` stopped development as of Sep. 2023, so I didn't consider a good idea to use its interface, which should have worked for strings/text.

Below a tenative design:
```python
class BaseTransform:
    def __call__(self, x: str | torch.Tensor ) -> str | torch.Tensor:
        # to be implemented
        pass
    
    def __repr__(self) -> str:
        return f"{self.__class__.__name__}()"
```
* `x` can be either `str`, for transformations that still operate with raw strings or `torch.Tensor`;
* similarly, `__call__:output` can be either `str` or `torch.Tensor`.

Without going into too much detail, but to explain why input and output type may be both... Some transformations we currently have are: (i) `str -> str`, e.g. when we augment data by adding the reverse complement of an amino acid sequence; (ii) others are `str -> torch.Tensor`, e.g. when we need to transform the raw strings into their numerical representation; and (iii) `torch.Tensor -> torch.Tensor`, e.g. when we apply masking.
